### PR TITLE
pages,routes and auth

### PR DESCRIPTION
**Pull request by deekshithmd**
Added pages, routes and test auth for video library. Please review it provide your comments

Main files to check:
components : Only few have code to check
contexts : initial data updation
App.js
services : api calls

### CODE REVIEW

File: src/components/ExploreVideo/explorevideo.css [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+.video-card img {
+    width: 100%;
+    height: 100%;
+}
+
+/* .video-details{
+    padding: 5px 9px;
+} */
```
Comment (saurabhchirde) [inline src/components/ExploreVideo/explorevideo.css]: remove comments

File: src/components/Home/home.css [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+.arrow-down {
+    font-size: 3rem;
+}
+
+.banner-btn {
+    padding: 1rem 0.5rem;
+    font-size: 2rem;
+    background-color: #09cc02;
```
Comment (saurabhchirde) [inline src/components/Home/home.css]: variable for color

File: src/components/Home/home.css [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    border-radius: 5px;
+}
+
+.banner-btn:hover {
+    background-color: #19a514;
```
Comment (saurabhchirde) [inline src/components/Home/home.css]: here as well

**Review by saurabhchirde — APPROVED** [review]
Could find any issue, your code is clean and easy to read. 
You can proceed to merge.

File: src/App.css [hunk 4]
```diff
@@ -1,51 +0,0 @@
...
-}
-
-@media (prefers-reduced-motion: no-preference) {
-  .App-logo {
-    animation: App-logo-spin infinite 20s linear;
-  }
-}
-
-.App-header {
-  background-color: #22201c;
```
Comment (ShrutiLakhani) [inline src/App.css:17]: Please try and use color variables wherever applicable

File: src/components/Home/home.css (hunk 3 above)
Comment (deekshithmd) [reply src/components/Home/home.css]: Thank you for pointing out...Thank you for PR review

**Review by ShrutiLakhani — APPROVED** [review]
Your code looks clean and easily readable. Good work!!!

File: src/App.css (hunk 4 above)
Comment (deekshithmd) [reply src/App.css:17]: Sure, Thank you for review

File: src/components/SideBar/SideBar.jsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import "./sidebar.css";
+import { Link } from "react-router-dom";
+export const SideBar = () => {
+  return (
+    <ul className="sidebar-container list-style-none">
```
Comment (BALAJI141031) [inline src/components/SideBar/SideBar.jsx:46]: @deekshithmd  here <link> element is repetitive so you can create one component for it and for path you can use array and you can iterate over array to give respective paths.
correct me if what i'm saying if not possible!!

File: src/components/Signup/Signup.jsx [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+              placeholder="******"
+              onMouseLeave={checkPassword}
+              required
+            />
+          </div>
+          <div className="form-input-message text-justify margin-l">
+            {password === false && (
+              <span className="input-validation error-msg">
+                <span className="message-icon">
+                  <i className="fa fa-exclamation-circle"></i>
```
Comment (mayuresh-tech) [inline src/components/Signup/Signup.jsx:118]: nice that you added validation too

File: src/components/Footer/Footer.jsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+          </a>
+        </li>
+        <li className="list-inline-item">
+          <a
+            href="http://twitter.com/deekshith_md"
+            target=" _blank"
+            rel="noreferrer"
+            className="nav-icon-link link-style-none"
+          >
+            <i className="fab fa-twitter nav-icon"></i>
```
Comment (mayuresh-tech) [inline src/components/Footer/Footer.jsx:38]: You can create a separate component for each link button.
And while calling you can pass github/twitter/linkedin icon and link

**Review by mayuresh-tech — APPROVED** [review]
good to merge

File: src/components/Footer/Footer.jsx (hunk 7 above)
Comment (deekshithmd) [reply src/components/Footer/Footer.jsx:38]: ok, Thank you for PR review

File: src/components/SideBar/SideBar.jsx (hunk 5 above)
Comment (deekshithmd) [reply src/components/SideBar/SideBar.jsx:46]: I will check, Thank you for suggesting

Comment (deekshithmd) [reply src/components/SideBar/SideBar.jsx:46]: It is possible by creating object of item data like pagename, icon class and path.

**Pull request merged**